packages/runtime-common/realm.ts

```diff
@@ -574,7 +575,12 @@ export class Realm {
 
   async #startup() {
     await Promise.resolve();
-    await this.#realmIndexUpdater.run();
+    let isNewIndex = await this.#realmIndexUpdater.isNewIndex();
+    let promise = this.#realmIndexUpdater.run();
+    if (isNewIndex) {
+      // we only await the full indexing at boot if this is a brand new index
+      await promise;
+    }
```

Comment:
same comment as above--why would this be marked as a diff? perhaps you need to merge upstream?